### PR TITLE
[ws-man-bridge] Setup redis WEB-595

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Setups redis, without actually using it. This PR ensures that we have the right config in place and can connect.

## Related Issue(s)

Fixes #

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - mp-ws-man-b7ae45b4fd</li>
	<li><b>🔗 URL</b> - <a href="https://mp-ws-man-b7ae45b4fd.preview.gitpod-dev.com/workspaces" target="_blank">mp-ws-man-b7ae45b4fd.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - mp-ws-man-bridge-redis-gha.12614</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

File: components/ws-manager-bridge/src/redis/client.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { inject, injectable, postConstruct } from "inversify";
+import { Redis } from "ioredis";
+import { Configuration } from "../config";
+
+@injectable()
+export class RedisClient {
```
Comment (easyCZ) [inline components/ws-manager-bridge/src/redis/client.ts:12]: I've copied this from `server`. How do we want to share this `Client` implementation between `server` and `ws-man-bridge`?

It's worth noting that this current implementation won't allow a given component to be both a subscriber, and publisher in pubsub. To have that, we'd need to create two distinct instances of the `client`. However, we don't yet have this use-case.

**Review by svenefftinge — APPROVED** [review]
lgtm

File: yarn.lock [hunk 2]
```diff
@@ -11332,14 +11378,26 @@ minimalistic-crypto-utils@^1.0.1:
...
+minimatch@^3.0.4:
   version "3.0.5"
   resolved "https://registry.yarnpkg.com/minimatch/-/minimatch-3.0.5.tgz#4da8f1290ee0f0f8e83d60ca69f8f134068604a3"
   integrity sha512-tUpxzX0VAzJHjLu0xUfFv1gwVp9ba3IOuRAVH2EGuRW8a5emA2FlACLqiT/lDVtS1W+TGNwqz3sWaNyLgDJWuw==
   dependencies:
     brace-expansion "^1.1.7"
 
-minimist@0.0.8, minimist@1.2.8, minimist@^1.1.0, minimist@^1.1.1, minimist@^1.2.0, minimist@^1.2.5:
+minimist@0.0.8:
+  version "0.0.8"
```
Comment (geropl) [inline yarn.lock:11396]: And this one as well :upside_down_face:

File: yarn.lock [hunk 3]
```diff
@@ -10624,7 +10624,23 @@ jsonify@~0.0.0:
   resolved "https://registry.npmjs.org/jsonify/-/jsonify-0.0.0.tgz"
   integrity sha1-LHS27kHZPKUbe1qu6PUDYx0lKnM=
 
-jsonwebtoken@^8.5.1, jsonwebtoken@^9.0.0:
+jsonwebtoken@^8.5.1:
+  version "8.5.1"
```
[the remark this answers is not in the document]
Comment (easyCZ) [reply yarn.lock:10628]: Sure!

[the remark this answers is not in the document]
Comment (easyCZ) [reply yarn.lock]: Ran `yarn install` from root. Can you see if it looks good now?

[the remark this answers is not in the document]
Comment (easyCZ) [reply yarn.lock]: Interestingly, it removed all yarn changes now :P

File: components/ws-manager-bridge/src/redis/client.ts (hunk 1 above)
Comment (easyCZ) [reply components/ws-manager-bridge/src/redis/client.ts:12]: I'll keep it here as-is for now

File: yarn.lock (hunk 3 above)
[the remark this answers is not in the document]
Comment (geropl) [reply yarn.lock]: That's actually expected, as we already import it inside the same workspace elsewhere (server). Running it locally just make it ignore [the resolutions](https://github.com/gitpod-io/gitpod/blob/cc52d25280cbcd35fe8c250b2da6c8a2852babc9/package.json#L14-L18).

**Pull request merged**